### ISSUE TITLE
Old libsdl version for Windows binaries

### ISSUE THREAD

**Issue by JNechaevsky** [fabiangreffrath/crispy-doom#188]
I have noticed, that Windows binaries still using SDL version 1.2.14, while current version is 1.2.15 (in both released and latest builds). Maybe it is worth to update SDL.dll to latest version, changelog can be found here: https://www.libsdl.org/release/changes-1.2.html

Also, SDL libs at latest.c-d.org are compiled as debug version (or with debug symbols, not sure how it sounds correct), which makes game sounds a bit jerky. It is not hard to hear the difference: just try to run released 4.0 and latest dev build - you will notice difference of the game sounds. In 4.0 everything is okay, but dev build have something like... frequency mismatch.

You are possibly remember, few month ago I have discovered same problem with Chocolate SDL2 dev builds, but I can't find that issue in Chocolate tracker. AFAIR, Simon have changed SDL libs from compiled versions to downloaded from the Internet.

_Suggestion №66_

**Comment by JNechaevsky** [comment]
Little update: hmm, even in case of using DLLs from released 4.0 in dev build, sound still jerking. Must this be because of different build environments? Any way, that's really not a big problem, since dev builds are for testing purposes only.

**Comment by fabiangreffrath** [comment]
I think this is because the dev builds are built without libsamplerate support. This would have nothing to do with SDL.

But I agree that upgrading SDL to the latest versions is a good idea (I didn't even know I was using an outdated version).

**Comment by JNechaevsky** [comment]
Fabian, important note about dev builds from latest.c-d.org!
Binary archive is missing following .DLLs:

- libgcc_s_dw2-1.dll
- libmad-0.dll
- libwinpthread-1.dll

They can be copied from released version 4.0 archive, but without them it is unable to run dev build.

**Comment by fabiangreffrath** [comment]
The dev builds are not done by myself but courtesy of @fragglet .

Are you sure you didn't mix up the libraries? I cannot remember the SDL_Mixer library from the dev builds depends on libmad, for example.

**Comment by Zodomaniac** [comment]
@JNechaevsky most likely you have SDL_Mixer.dll that is in fact renamed libsdl_mixer...dll from the Music Pack.

**Comment by JNechaevsky** [comment]
Everything is okay with today's latest build. Hmm.. Strange. Looks likes I've really messed up something yesterday.

**Comment by Zodomaniac** [comment]
I'll put a comment here as a reminder, for the newer version is available and should be introduced,

**Comment by fabiangreffrath** [comment]
I have updated all DLLs in the current Crispy Doom 4.1 release. Could you please confirm that these are now the most recent ones?

**Comment by JNechaevsky** [comment]
Checked out:

| DLL | Crispy | Latest SDL |
| ----|--------|----------- |
| SDL.dll | 1.2.14 | [1.2.15](https://www.libsdl.org/release/changes-1.2.html) ❗️ |
| libSDL_mixer-1-2-0.dll | 1.2.12 | [1.2.12](https://www.libsdl.org/projects/SDL_mixer/release-1.2.html) |
| SDL_net.dll | 1.2.8 | [1.2.8](https://www.libsdl.org/projects/SDL_net/release-1.2.html) |

Main SDL.dll is still outdated, please take a look at my first post.

**Comment by Zodomaniac** [comment]
@fabiangreffrath Please make the 4.2 release shortly with main SDL updated and DOS references removed) The https://github.com/fabiangreffrath/crispy-doom/issues/207 still needs testing, so I moved its milestone to 4.3.

**Comment by fabiangreffrath** [comment]
I have just checked, and the SDL.DLL library that i have bundled with the Crispy Doom 4.1 release reports itself as 1.2.15. Please try again with the attached program that I have copy-pasted from the SDL wiki (source included):

[sdlver.zip](https://github.com/fabiangreffrath/crispy-doom/files/1078747/sdlver.zip)

Now, the question remains why this library reports itself as SDL 1.2.14 in the Windows properties dialog. I have downloaded the SDL 1.2.15 sources from the SDL homepage, unpacked it and guess what I have found in `src/main/win32/version.rc`? ;)

**Comment by JNechaevsky** [comment]
Oh dang!
```
src/main/win32/version.rc
 FILEVERSION 1,2,14,0
 PRODUCTVERSION 1,2,14,0
```

Hovewer,
```
configure.in
 SDL_MAJOR_VERSION=1
 SDL_MINOR_VERSION=2
 SDL_MICRO_VERSION=15
```

Sdlver.exe saying:
```
We were compiled against SDL version 1.2.15.
We are linking against SDL version 1.2.15.
```

Guess, mr. Lantinga have forgotten to update version number in some source files. 
Alright then, let's just consider this as false alarm. 😎

**Comment by fabiangreffrath** [comment]
> Please make the 4.2 release shortly with main SDL updated and DOS references removed) 

Na, the SDL update is moot and the DOS references are purely cosmetical. It takes me about two hours to get a relase ready with booting into windows, compiling, updating the homepage here and the wiki there. I'd rather keep the established ~3 months cycle instead of releasing micro updates for low-priority bugs. ;)

> Just like you released 3.2 in a couple of days after 3.1 ;) 

But 3.1 contained a *major* bug that required re-compilation.

> The #207 still needs testing, so I moved its milestone to 4.3.

I am pretty confident that I'll get this into shape until the next release. ;)
